site/server/app/com/fortysevendeg/exercises/persistence/domain/UserProgressQueries.scala

```diff
@@ -5,47 +5,55 @@
 
 package com.fortysevendeg.exercises.persistence.domain
 
+import shared.UserProgress
+import doobie.imports._
+
 object SaveUserProgress {
 
-  sealed trait ExerciseType
+  sealed abstract class ExerciseType extends Product with Serializable
 
   case object Koans extends ExerciseType
 
   case object Other extends ExerciseType
 
   case class Request(
-    userId:       Long,
-    libraryName:  String,
-    sectionName:  String,
-    method:       String,
-    version:      Int,
-    exerciseType: ExerciseType   = Other,
-    args:         Option[String],
-    succeeded:    Boolean
-  )
+      userId:       Long,
+      libraryName:  String,
+      sectionName:  String,
+      method:       String,
+      version:      Int,
+      exerciseType: ExerciseType   = Other,
```

Comment:
Do we need to specify a default value here? If anything it would be Koan style by default.